x/group/keeper/msg_server.go

```diff
@@ -0,0 +1,810 @@
+			totalWeight, err = totalWeight.Add(newMemberWeight)
+			if err != nil {
+				return err
+			}
+		}
+		// Update group in the groupTable.
+		g.TotalWeight = totalWeight.String()
+		g.Version++
+		return k.groupTable.Update(ctx.KVStore(k.key), g.GroupId, g)
+	}
+
+	err := k.doUpdateGroup(ctx, req, action, "members updated")
+	if err != nil {
+		return nil, err
+	}
+
+	return &group.MsgUpdateGroupMembersResponse{}, nil
+}
+
+func (k Keeper) UpdateGroupAdmin(goCtx context.Context, req *group.MsgUpdateGroupAdmin) (*group.MsgUpdateGroupAdminResponse, error) {
+	ctx := types.UnwrapSDKContext(goCtx)
+	action := func(g *group.GroupInfo) error {
+		g.Admin = req.NewAdmin
+		g.Version++
+
+		return k.groupTable.Update(ctx.KVStore(k.key), g.GroupId, g)
+	}
+
+	err := k.doUpdateGroup(ctx, req, action, "admin updated")
+	if err != nil {
+		return nil, err
+	}
+
+	return &group.MsgUpdateGroupAdminResponse{}, nil
+}
+
+func (k Keeper) UpdateGroupMetadata(goCtx context.Context, req *group.MsgUpdateGroupMetadata) (*group.MsgUpdateGroupMetadataResponse, error) {
+	ctx := types.UnwrapSDKContext(goCtx)
+	action := func(g *group.GroupInfo) error {
+		g.Metadata = req.Metadata
+		g.Version++
+		return k.groupTable.Update(ctx.KVStore(k.key), g.GroupId, g)
+	}
+
+	if err := assertMetadataLength(req.Metadata, "group metadata"); err != nil {
+		return nil, err
+	}
+
+	err := k.doUpdateGroup(ctx, req, action, "metadata updated")
+	if err != nil {
+		return nil, err
+	}
+
+	return &group.MsgUpdateGroupMetadataResponse{}, nil
+}
+
+func (k Keeper) CreateGroupAccount(goCtx context.Context, req *group.MsgCreateGroupAccount) (*group.MsgCreateGroupAccountResponse, error) {
+	ctx := types.UnwrapSDKContext(goCtx)
+	admin, err := sdk.AccAddressFromBech32(req.GetAdmin())
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "request admin")
+	}
+	policy := req.GetDecisionPolicy()
+	groupID := req.GetGroupID()
+	metadata := req.GetMetadata()
+
+	if err := assertMetadataLength(metadata, "group account metadata"); err != nil {
+		return nil, err
+	}
+
+	g, err := k.getGroupInfo(ctx.Context(), groupID)
+	if err != nil {
+		return nil, err
+	}
+	groupAdmin, err := sdk.AccAddressFromBech32(g.Admin)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "group admin")
+	}
+	// Only current group admin is authorized to create a group account for this
+	if !groupAdmin.Equals(admin) {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "not group admin")
+	}
+
+	// Generate group account address.
+	var accountAddr sdk.AccAddress
+	var accountDerivationKey []byte
+	// loop here in the rare case of a collision
+	for {
+		nextAccVal := k.groupAccountSeq.NextVal(ctx.KVStore(k.key))
+		buf := bytes.NewBuffer(nil)
+		err = binary.Write(buf, binary.LittleEndian, nextAccVal)
+		if err != nil {
+			return nil, err
+		}
+
+		accountDerivationKey = buf.Bytes()
+		accountAddr := group.AccountCondition(k.groupAccountSeq.NextVal(ctx.KVStore(k.key))).Address()
```

Comment:
this needs to use `types/address` instead to derive the group module address (use `Module` function to get that) with the `accountDerivationKey` (see `Derive` function)
see https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-028-public-key-addresses.md#derived-addresses for more info about ADR 028